server/backend/api/app/crud/crud_user_role.py

```diff
@@ -117,12 +117,18 @@ def find_or_create(db: Session, user_id: int, role_id: int):
         db.flush()
         LOGGER.debug(f"New FamUserRoleXref added for {new_fam_user_role.__dict__}")
         return new_fam_user_role
-
-    LOGGER.debug(
-        "FamUserRoleXref already exists with id: " +
-        f"{fam_user_role_xref.user_role_xref_id}."
-    )
-    return fam_user_role_xref
+    else:
+        LOGGER.debug(
+            "FamUserRoleXref already exists with id: " +
+            f"{fam_user_role_xref.user_role_xref_id}."
+        )
+        error_msg = f'{"Role already assigned to user."}'
+        crud_utils.raise_http_exception(HTTPStatus.CONFLICT, error_msg)
```

Comment:
Hi Nick,
It's nice to raise exception/warning when intended user is already assigned the same role.
But on the second thought, this is in a function "**find_or_create()**" - so its original intention was to find the record or else create new one and return object for the rest of the caller function.
Raising exception inside this function would **destroy** its original intention ('find' purpose).
That was the story before.

Now our purpose is changed to let the frontend knows that (by raising exception with message to the frontend) user has been assigned. So I am thinking this "**find_or_create()**" can be removed or simply rename it to something similar like "**create_user_role(db, user_id, role_id)**" and you don't even need to raise exception inside this "create" function.
But you then can move your raise exception logic outside to the main caller function, using "get_use_role_by_user_id_and_role_id()" with if statement to check if user-role exists before calling "create" funcgtion.

What do you think? If it is not clear to you, we can have short chat.

Reply:
I think you raised a good point. I am going to adjust this function.